From 6d329337f6701fe090602c6f8ca373250eb91b82 Mon Sep 17 00:00:00 2001
From: Alan Modra <amodra@gmail.com>
Date: Tue, 30 Jan 2024 09:38:56 +1030
Subject: [PATCH 002/115] PR31314, chew crashing on use of uninitialized value

The "drop" call in wrap_comment already increments pc.  Defining DOCDD
in proto.str is a warning fix.

	PR 31314
	* chew.c (wrap_comment): Don't increment pc.
	* proto.str (DOCDD): Define.

(cherry picked from commit e175a2fc60cb9709c4461cdd8596ae05e529d67b)
---
 bfd/doc/chew.c    | 2 --
 bfd/doc/proto.str | 1 +
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/bfd/doc/chew.c b/bfd/doc/chew.c
index 28c8072c272..842d4153005 100644
--- a/bfd/doc/chew.c
+++ b/bfd/doc/chew.c
@@ -541,8 +541,6 @@ wrap_comment (void)
 
   overwrite_string (tos - 1, &out);
   drop ();
-
-  pc++;
 }
 
 /* Mod tos so that only lines with leading dots remain */
diff --git a/bfd/doc/proto.str b/bfd/doc/proto.str
index ee06c31fa7c..9a56397d719 100644
--- a/bfd/doc/proto.str
+++ b/bfd/doc/proto.str
@@ -166,6 +166,7 @@
 : SUBSUBSECTION ignore ;
 : INODE ignore ;
 : DESCRIPTION ignore ;
+: DOCDD ignore ;
 : ENDDD EXTERNAL ;
 : FUNCTION EXTERNAL ;
 : INTERNAL_FUNCTION INTERNAL ;
-- 
2.44.0

